### PR TITLE
Add domain upsell nudge for plan owners with domain

### PR DESCRIPTION
**Pull request by eltongo**
**Changes proposed in this Pull Request**

Adds the domain upsell nudge, discussed here: p99Zz8-K5-p2

Depends on D32926-code

**Testing instructions**

- Verify the nudge only appears to sites with a plan and at least one domain, if there are no other notices/nudges
- Make sure it displays the correct data (hack the backend to return custom price and sale data and verify the nudge displays correct information)

### CODE REVIEW

**Review by daledupreez — COMMENTED** [review]
My comments are up.

Based on my experience, we're really aiming for a small group -- this notice won't appear to someone looking at a site without a plan, even if the user has other paid plans. That may be worth flagging as part of the general discussions.

File: client/my-sites/current-site/notice.jsx [hunk 1]
```diff
@@ -95,6 +102,85 @@ export class SiteNotice extends React.Component {
...
 
+	domainUpsellNudge() {
+		if ( ! this.props.isPlanOwnerAndHasDomains ) {
+			return null;
+		}
+
+		const { currencyCode, productsList, translate } = this.props;
+
+		const priceAndSaleInfo = transform( productsList, function( result, value, key ) {
+			if ( value.is_domain_registration && value.available && key !== 'domain_reg' ) {
```
[the remark this answers is not in the document]
Comment (eltongo) [reply client/my-sites/current-site/notice.jsx]: That's right. Good one!

File: client/my-sites/current-site/notice.jsx [hunk 2]
```diff
@@ -95,6 +102,85 @@ export class SiteNotice extends React.Component {
...
+				const minRegularPrice = get( result, 'minRegularPrice', regularPrice );
+				result.minRegularPrice = Math.min( minRegularPrice, regularPrice );
+
+				const salePrice = getUnformattedDomainSalePrice( key, productsList );
+				if ( salePrice ) {
+					const minSalePrice = get( result, 'minSalePrice', salePrice );
+					result.minSalePrice = Math.min( minSalePrice, salePrice );
+
+					if ( ! result.saleTlds ) {
+						result.saleTlds = [];
```
[the remark this answers is not in the document]
Comment (eltongo) [reply client/my-sites/current-site/notice.jsx]: I don't think this piece of code would be particularly risky, but I think we may shave off a couple of bytes if we initialize it :)

File: client/my-sites/current-site/notice.jsx [hunk 3]
```diff
@@ -95,6 +102,85 @@ export class SiteNotice extends React.Component {
...
+		} else {
+			noticeText = translate( 'Get a domain starting at %(minDomainPrice)s', {
+				args: {
+					minDomainPrice: formatCurrency( priceAndSaleInfo.minRegularPrice, currencyCode ),
+				},
+			} );
+		}
+
+		return (
+			<Notice isCompact status="is-success" icon="info-outline" text={ noticeText }>
```
[the remark this answers is not in the document]
Comment (eltongo) [reply client/my-sites/current-site/notice.jsx]: I went with the more popular `Notice`. To me all green means free, which is not the case in this instance.

**Comment by eltongo** [comment]
> this notice won't appear to someone looking at a site without a plan, even if the user has other paid plans

I may be wrong on this one, but I think there will be other nudges/notices present if the chosen site is on a free plan.

File: client/my-sites/current-site/notice.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (daledupreez) [reply client/my-sites/current-site/notice.jsx]: I think the text in the free-to-paid nudge is misleading -- the _domain_ is free, but the _upgrade_ is not! 😉

I did some digging -- based on #20846, it looks like `SidebarBanner` was added especially for upsell nudges.

**Comment by daledupreez** [comment]
> I may be wrong on this one, but I think there will be other nudges/notices present if the chosen site is on a free plan.

That is what I was seeing! 😁 
Basically, I have site A with a custom domain and a Business plan, and I have site B that is free without a domain or a plan. When I look at site A, I see the domain-related nudge. However, when I look at site B, I don't see a domain-related nudge, I see only the free-to-paid nudge.

I _think_ that's what we want, but think it may be worth confirming that (unless you have already done that!).

File: client/my-sites/current-site/notice.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (eltongo) [reply client/my-sites/current-site/notice.jsx]: > I think the text in the free-to-paid nudge is misleading -- the domain is free, but the upgrade is not!

Yeah, I meant green as a color, not the specific wording of the nudge in your screenshot :). This merits more discussion though.

**Comment by daledupreez** [comment]
Looks good to me, including the banner triggering the right CTA events getting sent to the server. :+1:
We may want Del/Kamen to take a look, too.

**Review by daledupreez — COMMENTED** [review]
Looks good to me. :)

**Comment by Stojdza** [comment]
@daledupreez I added `Needs e2e Testing` label to spin up tests against this PR. I think this change can affect some of the e2e tests.

**Review by delputnam — APPROVED** [review]
This LGTM and works for me.  Nice job.

**Pull request merged**